news/_posts/2019-02-27-we-were-at-the-cms-security-summit-with-google.md

```diff
@@ -0,0 +1,103 @@
+
+In the Prestashop Addons Marketplace security process, each submitted module is thoroughly reviewed by our fantastic team. Each payment module is also [pentested](https://www.owasp.org/index.php/Web_Application_Penetration_Testing), even the ones not developed by the PrestaShop team. 
+
+Of course, we all know that even fantastic humans are error-prone: that is why PrestaShop invests in continually improving the automated [Module validator](https://validator.prestashop.com/). Other CMS are also considering  static code analysis tools like [scrutinizer](https://scrutinizer-ci.com/) or [snyk](https://snyk.io/). That is maybe something to add to our validation tools.
+
+We are currently discussing about adding PHPStan and PHPQA to our modules, which are two really great tools to improve code quality.
+
+Another improvement currently under consideration is to cryptographically sign modules or themes sold on the PrestaShop Marketplace, so that merchants can make sure they are legitimate and haven’t been modified by an attacker.
+
+
+## Vulnerability disclosures & rewards
+
+This is known, there is no Bug Bounty program for the PrestaShop project… yet!
+But it can be a real issue:
+
+- it can be hard for contributors to spot where they must go to report security issues
+- some researchers need rewards to feel motivated enought to create a report
+
+What kind of reward would you want? A tee-shirt? Money? Preston Plushies? Goodies?
+
+Drupal is kind of the leader here: they have a Bug Bounty program, documentation about [How to report a security issue](https://www.drupal.org/security-team/report-issue) and a [CVE history](https://www.drupal.org/security). 
+Drupal can be a good source of inspiration for PrestaShop here. 
+
+Moreover, since the group that creates [PHP interoperability standards](https://www.php-fig.org) has reopened what they started for PSR-9 (Security Advisories) and PSR-10 (Security Reporting Process), it appears to be a good reason to take some time between two code reviews, think about a bug bounty program and write documentation about it.
+
+
+## One click upgrade or autoupgrade?
+
+The best way to have a secure system is to stay up-to-date. Many people don’t care about upgrades, but they don’t realize that security fixes need to be applied as fast as possible, or else their shop is at risk!
+ 
+Seeing things exclusively through an economic prism can be misleading: yes, upgrading may cost you money now. But think about what could be lost if an intruder comes, injects an XSS and steals your clients’ information, or adds a script that shuts shop off or deletes everything you had!
+
+Well, how to update a CMS when a lot of modules or a custom theme are already installed? How to avoid breaking something? How to be sure that users and developers are following the best practices?
```

Comment:
This paragraph needs some kind of answer

Reply:
But nobody have answered it yet :/
These are real questions. We need feedback from our community and people which are afraid because of bad upgrade.

Reply:
```suggestion
But, how to update a CMS when a lot of modules or a custom theme are already installed? How to avoid breaking something that we are not aware of? How to be sure that users and developers are following the best practices?

These are very difficult questions to which we still don't have the answer. But we are working on it. If you encounter problems during upgrading your shop, send us your feedback!
```